### PR TITLE
systemlog indexer and api

### PR DESCRIPTION
**Pull request by Frankonly**

### CODE REVIEW

File: chainservice/chainservice.go [hunk 1]
```diff
@@ -165,12 +179,8 @@ func New(
 		cfg.DB.DbPath = cfg.Chain.ChainDBPath
 		kvStore = db.NewBoltDB(cfg.DB)
 	}
-	var dao blockdao.BlockDAO
-	if gateway && !cfg.Chain.EnableAsyncIndexWrite {
-		dao = blockdao.NewBlockDAO(kvStore, []blockdao.BlockIndexer{indexer}, cfg.Chain.CompressBlock, cfg.DB)
-	} else {
-		dao = blockdao.NewBlockDAO(kvStore, nil, cfg.Chain.CompressBlock, cfg.DB)
-	}
+	dao := blockdao.NewBlockDAO(kvStore, indexers, cfg.Chain.CompressBlock, cfg.DB)
```
Comment (CoderZhi) [inline chainservice/chainservice.go]: indexers are appended to blockdao in sync mode. In async mode, we need to add the indexers as subscriber. So you need to revert this change.

File: systemlog/systemlog.go [hunk 2]
```diff
@@ -10,6 +10,8 @@ import (
 	"context"
 	"sync"
 
+	"github.com/iotexproject/iotex-proto/golang/iotextypes"
```
Comment (CoderZhi) [inline systemlog/systemlog.go]: wrong group

File: chainservice/chainservice.go (hunk 1 above)
Comment (Frankonly) [reply chainservice/chainservice.go]: Get.

Comment (Frankonly) [reply chainservice/chainservice.go]: I handle this at line 163
```
if !cfg.Chain.EnableAsyncIndexWrite {
	indexers = append(indexers, indexer)
}
```

Comment (Frankonly) [reply chainservice/chainservice.go]: Or we need to make systemlog indexer as subsrciber?

File: chainservice/chainservice.go [hunk 3]
```diff
@@ -167,11 +183,8 @@ func New(
 		kvStore = db.NewBoltDB(cfg.DB)
 	}
 	var dao blockdao.BlockDAO
-	if gateway && !cfg.Chain.EnableAsyncIndexWrite {
-		dao = blockdao.NewBlockDAO(kvStore, []blockdao.BlockIndexer{indexer}, cfg.Chain.CompressBlock, cfg.DB)
-	} else {
-		dao = blockdao.NewBlockDAO(kvStore, nil, cfg.Chain.CompressBlock, cfg.DB)
-	}
+	dao = blockdao.NewBlockDAO(kvStore, indexers, cfg.Chain.CompressBlock, cfg.DB)
```
Comment (CoderZhi) [inline chainservice/chainservice.go]: one line

**Comment by CoderZhi** [comment]
It is better to have system log indexer and the action indexer sharing the same config, EnableAsyncIndexing.

**Comment by Frankonly** [comment]
#1346
